src/qibosoq/programs/base.py

```diff
@@ -223,15 +222,11 @@ def body(self):
 
         self.wait_all()
 
-    # pylint: disable=unexpected-keyword-arg, arguments-renamed
-    def acquire(
+    def perform_experiment(
```

Comment:
Since you changed the name, maybe you should take the chance to also update the docstring with a more explicit one.